### ISSUE TITLE
MagicBrush does not work at all as good as InstructPix2Pix

### ISSUE THREAD

**Issue by sanchit88** [OSU-NLP-Group/MagicBrush#5]
Hi Authors, recently I am doing a comparison between MagicBrush (fine tuned one) and InstructPix2Pix. To my surprise MagicBrush is not working that good as compared to InstructPix2Pix. It is even quite off compared to it. I am really wondering if this is really the case or am I missing something. I have tested with about 25 images, with simple instructions and also from not so big masked areas. Can you please help me here what is going on? Could it be possible that those 10K images are way too small dataset and model has simply overfitted to them. It is really not working as per the qualitative studies mentioned in the paper. Thanks a lot!

**Comment by drogozhang** [comment]
Hi,

Thanks for your question!
1. We fine-tuned our model on the training set and tested it on another split, so it's not likely to be overfitting. The comparison in the paper was randomly conducted over examples on the test set. So I don't think InstructPix2Pix is better than the fine-tuned one. Granted, sometimes InstructPix2Pix does perform better on some edits, but this doesn't hold up when tested on more examples.

2. For your situation, I personally would suggest trying different instructions with the same editing intention. Our instructions are written by crowd workers and they may follow certain patterns. The same problem occurs in the original InstructPix2Pix; these two models may excel with different instructions.

3. We didn't extensively search the hyper-parameter space for fine-tuning, and I believe there are better recipes for it. You might want to try fine-tuning a better checkpoint :)

4. We are interested in the failure cases you're encountering and would like to help you so we can improve our models. Feel free to share these results with me at `drogozhang@gmail.com` if you're comfortable doing so.

**Comment by drogozhang** [comment]
Hi,

I also uploaded a better checkpoint fine-tuned on Magic Brush with different recipe (fewer epochs), namely `MagicBrush-epoch-52-step-4999.ckpt`.

You can follow the updated README.md to have a try :)

Let me know if you have any other questions.

**Comment by sanchit88** [comment]
Hi @drogozhang Thanks a lot for your kind reply. I am going to test your new checkpoint and compares it back with InstructPix2Pix. Yes, I do have up to 4 different instructions for an image (e.g., by changing verb or adjectives etc). At the moment, I found InstructPix2Pix to be more robust than over similar methods (including yours). 
Yes, I will send you also some test examples. Thanks a lot again for your openness. Really appreciated!

**Comment by drogozhang** [comment]
Thanks @sanchit88 for your valuable insights on InstructPix2Pix!
The robustness to instructions is an important problem, we will look into it and update the model :)

**Comment by drogozhang** [comment]
Close it for inactivity, feel free to re-open it :)
